sale_delivery_date/models/sale_order_line.py

```diff
@@ -43,129 +43,79 @@ def _prepare_procurement_values_commitment_date(self, res):
-                        partner,
-                        partner.order_delivery_cutoff_preference,
-                        self.order_id.warehouse_id,
-                    )
-                )
-            else:
-                _logger.warning(
-                    "No cutoff applied on order %s. %s time not applied"
-                    "on line %s."
-                    % (self.order_id, partner.order_delivery_cutoff_preference, self)
-                )
-            return date_planned
-        new_date_planned = self._get_utc_cutoff_datetime(
-            cutoff, date_planned, keep_same_day
-        )
-        _logger.debug(
-            "%s applied on order %s. Date planned for line %s"
-            " rescheduled from %s to %s"
-            % (
-                partner.order_delivery_cutoff_preference,
-                self.order_id,
-                self,
-                date_planned,
-                new_date_planned,
-            )
-        )
-        return new_date_planned
-
-    def _get_date_planned_with_warehouse_calendar(self, date_planned):
-        """Update the date planned based on the warehouse calendar if any."""
-        calendar = self.order_id.warehouse_id.calendar_id
-        if date_planned and calendar:
-            customer_lead, security_lead, workload = self._get_delays()
-            # plan_days() expect a number of days instead of a delay
-            workload_days = self._delay_to_days(workload)
-            # Add the workload, with respect to the wh calendar
-            date_planned = calendar.plan_days(
-                workload_days, date_planned, compute_leaves=True
-            )
-        return date_planned
+            return date
+        return self._get_utc_cutoff_datetime(cutoff, date, keep_same_day)
 
-    def _get_date_deadline_with_delivery_window(self, date_planned):
-        """Return 'date_deadline' according to customer's time windows.
+    @api.model
+    def _apply_workload(self, date, days, calendar=None):
+        if calendar:
+            return calendar.plan_days(days, date, compute_leaves=True)
+        return date
 
-        This computation is called only if no commitment_date is set and
-        if the customer's delivery time preference is "time windows".
-        It will return the effective delivery date by considering the next
-        preferred delivery time window of the customer.
-        """
-        # As the 'date_planned' is the date when the work can start on the
-        # transfer, we have to add the security lead of the company to get
-        # the date indicating when the transfer will be ready to be shipped.
-        # From this date we will be able to compute the preferred delivery date
-        # of the customer.
-        date_transfer_done = date_planned + timedelta(
+    @api.model
+    def _apply_delivery_window(self, date_planned, partner, calendar=None):
+        date_done = date_planned + timedelta(
             days=self.order_id.company_id.security_lead
```

Comment:
/!\ You are making use of `self.order_id` in an `@api.model` method, we should also send the security lead or the order as parameter then